src/main/java/hmim/eteam/rest/backend/api/IRoleResolver.java

```diff
@@ -0,0 +1,15 @@
+package hmim.eteam.rest.backend.api;
+
+import hmim.eteam.rest.backend.entity.user.AuthToken;
+import hmim.eteam.rest.backend.entity.user.SiteUser;
+import hmim.eteam.rest.backend.entity.user.UserRole;
+import org.springframework.lang.Nullable;
+
+import javax.validation.constraints.NotNull;
+import java.util.Optional;
+
+public interface IRoleResolver {
+    UserRole resolve(@NotNull String authTokenId, @Nullable Long courseId);
+    UserRole resolve(@NotNull Optional<AuthToken> authToken, @Nullable Long courseId);
```

Comment:
This method is kind of syntax sugar, that simplifies operations with repositories. Repositories return optionals, therefore to avoid spamming if-elses in requests code we use such strange construction.